conf/config.go

```diff
@@ -21,6 +21,8 @@ type Config struct {
 	ConstFns    map[string]reflect.Value
 	Visitors    []ast.Visitor
 	Functions   map[string]*builtin.Function
+
+	AllowReuseCommon bool // allow cache common sub-expr computed result, aimed to reuse already computed result
```

Comment:
For functions will make more sense a config like this:
```go
expr.PureFunc("fib")
```

Reply:
Similar to expr.ConstExpr

Reply:
> Similar to expr.ConstExpr
Yes, I already have added this function `expr.AllowReuseCommon()` in expr.go. 
https://github.com/zhuliquan/expr/blob/f84c5a98f555a187a861bcc94218d80d0fa2c7ed/expr.go#L127